### PR TITLE
Fixing fees added via old filter and items now added with per-item price not total amount

### PR DESCRIPTION
**Pull request by cklosowski**
#2667

### CODE REVIEW

**Comment by pippinsplugins** [comment]
@cklosowski Please add `@since` tags to all properties.

File: includes/payments/class-edd-payment.php [hunk 1]
```diff
@@ -33,39 +33,208 @@
...
+
+	/**
+	 * Array of global fees for this payment
+	 * @var array
+	 */
+	protected $fees = array();
+
+	/**
+	 * The sum of the fee amounts
+	 * @var integer
```
Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: This is a float.

File: includes/payments/class-edd-payment.php [hunk 2]
```diff
@@ -33,39 +33,208 @@
...
+	 * The Unique Payment Key
+	 * @var string
+	 */
+	protected $key = '';
+
+	/**
+	 * The total amount the payment is for
+	 * Includes items, taxes, fees, and discounts
+	 *
+	 * @var integer
```
Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: This is a float

File: includes/payments/class-edd-payment.php [hunk 3]
```diff
@@ -591,7 +787,7 @@ public function add_download( $download_id = 0, $args = array(), $options = arra
 		$defaults = array(
 			'quantity'    => 1,
 			'price_id'    => false,
-			'amount'      => false,
+			'item_price'  => false,
```
Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: `item_price` should probably default to either `0.00` or `''`

**Comment by pippinsplugins** [comment]
Fees look really good. Just one thing: it doesn't look like the fees handling in `add_download()` ensures the `download_id` is set on the fee.

If a fee is supplied in the `$args` array of `add_download()`, we should force it to include `download_id` and set that to the ID passed to `add_download()`

**Comment by cklosowski** [comment]
:+1: will get those done quick, should be easy

**Comment by cklosowski** [comment]
@pippinsplugins Ok these are ready for your look again.

**Pull request merged**